linen_examples/mnist/mnist_lib.py

```diff
@@ -148,30 +150,30 @@ def get_datasets():
   return train_ds, test_ds
 
 
-def train_and_evaluate(model_dir: str, num_epochs: int, batch_size: int,
-                       learning_rate: float, momentum: float):
+def train_and_evaluate(hp_config: ml_collections.ConfigDict, model_dir: str):
```

Comment:
I would vote for just `config`

and instead write `jax.config.enable_omnistaging()`

Reply:
Done.